File: eds-starter-onemany/src/main/java/ch/rasc/eds/starter/entity/setup/Uom.java
package ch.rasc.eds.starter.entity.setup;

import java.time.ZonedDateTime;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.NotBlank;

/**
 * Created by Administrator on 25/08/2015.
 */
import ch.rasc.edsutil.entity.AbstractPersistable;
import ch.rasc.extclassgenerator.Model;
import ch.rasc.extclassgenerator.ModelField;

@Entity
@Table(name = "Uom")
@Model(value = "Starter.model.setup.Uom", readMethod = "uomService.read",
        createMethod = "uomService.create", updateMethod = "uomService.update",
        destroyMethod = "uomService.destroy", paging = true, identifier = "negative")
public class Uom extends AbstractPersistable {

    @NotBlank(message = "{fieldrequired}")
    @Size(max = 255)
    private String uomName;


    @Size(max = 255)
    @ModelField(allowNull = true)
    private String notes;

    @ModelField(dateFormat = "time", persist = false)
    private ZonedDateTime lastUpdate;

    public String getUomName() {
        return this.uomName;
    }

    public void setUomName(String uomName) {
        this.uomName = uomName;
    }

    public String getNotes() {
        return this.notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public ZonedDateTime getLastUpdate() {
        return this.lastUpdate;
    }

    public void setLastUpdate(ZonedDateTime lastUpdate) {
        this.lastUpdate = lastUpdate;
    }
}

File: eds-starter-onemany/src/main/resources/static/app/model/setup/Uom.js
Ext.define("Starter.model.setup.Uom",
{
  extend : "Ext.data.Model",
  requires : [ "Ext.data.identifier.Negative", "Ext.data.proxy.Direct", "Ext.data.validator.Length" ],
  identifier : "negative",
  fields : [ {
    name : "uomName",
    type : "string",
    validators : [ {
      type : "length",
      min : 0,
      max : 255
    } ]
  }, {
    name : "notes",
    type : "string",
    allowNull : true,
    validators : [ {
      type : "length",
      min : 0,
      max : 255
    } ]
  }, {
    name : "lastUpdate",
    type : "date",
    dateFormat : "time",
    persist : false
  }, {
    name : "id",
    type : "integer",
    allowNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "uomService.read",
      create : "uomService.create",
      update : "uomService.update",
      destroy : "uomService.destroy"
    },
    reader : {
      rootProperty : "records"
    },
    writer : {
      writeAllFields : true
    }
  }
});